app/build.gradle

```diff
@@ -21,30 +21,31 @@ android {
 
 dependencies {
     implementation fileTree(dir: 'libs', include: ['*.jar'])
-    implementation 'androidx.appcompat:appcompat:1.1.0'
+    implementation 'androidx.appcompat:appcompat:1.3.1'
 
-    implementation 'com.google.firebase:firebase-core:11.0.4'
-    implementation 'com.google.firebase:firebase-auth:11.0.4'
-    implementation 'com.google.firebase:firebase-database:11.0.4'
-    implementation 'com.google.firebase:firebase-storage:11.0.4'
-    implementation 'com.google.android.gms:play-services-maps:11.0.4'
+    implementation 'com.google.firebase:firebase-core:19.0.1'
+    implementation 'com.google.firebase:firebase-auth:21.0.1'
+    implementation 'com.google.firebase:firebase-database:20.0.2'
+    implementation 'com.google.firebase:firebase-storage:20.0.0'
+    implementation 'com.google.android.gms:play-services-maps:17.0.1'
     implementation 'com.firebaseui:firebase-ui-database:0.4.0'
 
-    testImplementation 'junit:junit:4.12'
-    androidTestImplementation 'androidx.test.ext:junit:1.1.1'
-    androidTestImplementation 'androidx.test.espresso:espresso-core:3.2.0'
-    implementation 'com.google.android.material:material:1.0.0'
+    testImplementation 'junit:junit:4.13.2'
+    androidTestImplementation 'androidx.test.ext:junit:1.1.3'
+    androidTestImplementation 'androidx.test.espresso:espresso-core:3.4.0'
+    implementation 'com.google.android.material:material:1.4.0'
 
 
-    implementation 'com.google.android.gms:play-services-location:11.0.4'
-    implementation 'com.google.android.gms:play-services:11.0.4'
-    implementation 'com.squareup.picasso:picasso:2.5.2'
+    implementation 'com.google.android.gms:play-services-location:18.0.0'
+    //noinspection UseOfBundledGooglePlayServices
+    implementation 'com.google.android.gms:play-services:12.0.1'
+    implementation 'com.squareup.picasso:picasso:2.71828'
     implementation 'com.miguelcatalan:materialsearchview:1.4.0'
 
 
-    implementation 'androidx.constraintlayout:constraintlayout:1.1.3'
-    implementation 'androidx.multidex:multidex:2.0.0'
+    implementation 'androidx.constraintlayout:constraintlayout:2.1.1'
+    implementation 'androidx.multidex:multidex:2.0.1'
     implementation 'info.hoang8f:fbutton:1.0.5'
-    implementation 'androidx.mediarouter:mediarouter:1.0.0'
+    implementation 'androidx.mediarouter:mediarouter:1.2.5'
```

Comment:
Executing tasks: [:app:assembleDebug] in project D:\gitProjects\ReviewProject\WildCare

AGPBI: {"kind":"warning","text":"The specified Android SDK Build Tools version (29.0.2) is ignored, as it is below the minimum supported version (30.0.2) for Android Gradle Plugin 7.0.2.\nAndroid SDK Build Tools 30.0.2 will be used.\nTo suppress this warning, remove \"buildToolsVersion '29.0.2'\" from your build.gradle file, as each version of the Android Gradle Plugin now has a default version of the build tools.","sources":[{}]}
> Task :app:preBuild UP-TO-DATE
> Task :app:preDebugBuild UP-TO-DATE
> Task :app:compileDebugAidl NO-SOURCE
> Task :app:compileDebugRenderscript NO-SOURCE
> Task :app:generateDebugBuildConfig
> Task :app:javaPreCompileDebug
> Task :app:checkDebugAarMetadata FAILED
> Task :app:generateDebugResValues
> Task :app:generateDebugResources
> Task :app:processDebugGoogleServices
> Task :app:mergeDebugResources FAILED
> Task :app:createDebugCompatibleScreenManifests
> Task :app:extractDeepLinksDebug
> Task :app:processDebugMainManifest FAILED
> Task :app:mergeDebugNativeDebugMetadata NO-SOURCE
> Task :app:mergeDebugShaders
> Task :app:compileDebugShaders NO-SOURCE
> Task :app:generateDebugAssets UP-TO-DATE
> Task :app:mergeDebugAssets FAILED
> Task :app:processDebugJavaRes NO-SOURCE
> Task :app:mergeDebugJavaResource FAILED
> Task :app:checkDebugDuplicateClasses FAILED
> Task :app:desugarDebugFileDependencies FAILED
> Task :app:mergeDebugJniLibFolders
> Task :app:mergeDebugNativeLibs FAILED
> Task :app:validateSigningDebug
> Task :app:writeDebugAppMetadata
> Task :app:writeDebugSigningConfigVersions

FAILURE: Build completed with 8 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':app:checkDebugAarMetadata'.
> Could not resolve all files for configuration ':app:debugRuntimeClasspath'.
   > Could not find com.firebaseui:firebase-ui-database:0.4.0.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/com/firebaseui/firebase-ui-database/0.4.0/firebase-ui-database-0.4.0.pom
       - https://repo.maven.apache.org/maven2/com/firebaseui/firebase-ui-database/0.4.0/firebase-ui-database-0.4.0.pom
     Required by:
         project :app
   > Could not find com.miguelcatalan:materialsearchview:1.4.0.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/com/miguelcatalan/materialsearchview/1.4.0/materialsearchview-1.4.0.pom
       - https://repo.maven.apache.org/maven2/com/miguelcatalan/materialsearchview/1.4.0/materialsearchview-1.4.0.pom
     Required by:
         project :app

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':app:mergeDebugResources'.
> Could not resolve all files for configuration ':app:debugRuntimeClasspath'.
   > Could not find com.firebaseui:firebase-ui-database:0.4.0.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/com/firebaseui/firebase-ui-database/0.4.0/firebase-ui-database-0.4.0.pom
       - https://repo.maven.apache.org/maven2/com/firebaseui/firebase-ui-database/0.4.0/firebase-ui-database-0.4.0.pom
     Required by:
         project :app
   > Could not find com.miguelcatalan:materialsearchview:1.4.0.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/com/miguelcatalan/materialsearchview/1.4.0/materialsearchview-1.4.0.pom
       - https://repo.maven.apache.org/maven2/com/miguelcatalan/materialsearchview/1.4.0/materialsearchview-1.4.0.pom
     Required by:
         project :app

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
==============================================================================

3: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':app:processDebugMainManifest'.
> Could not resolve all files for configuration ':app:debugRuntimeClasspath'.
   > Could not find com.firebaseui:firebase-ui-database:0.4.0.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/com/firebaseui/firebase-ui-database/0.4.0/firebase-ui-database-0.4.0.pom
       - https://repo.maven.apache.org/maven2/com/firebaseui/firebase-ui-database/0.4.0/firebase-ui-database-0.4.0.pom
     Required by:
         project :app
   > Could not find com.miguelcatalan:materialsearchview:1.4.0.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/com/miguelcatalan/materialsearchview/1.4.0/materialsearchview-1.4.0.pom
       - https://repo.maven.apache.org/maven2/com/miguelcatalan/materialsearchview/1.4.0/materialsearchview-1.4.0.pom
     Required by:
         project :app

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
==============================================================================

4: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':app:mergeDebugAssets'.
> Could not resolve all files for configuration ':app:debugRuntimeClasspath'.
   > Could not find com.firebaseui:firebase-ui-database:0.4.0.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/com/firebaseui/firebase-ui-database/0.4.0/firebase-ui-database-0.4.0.pom
       - https://repo.maven.apache.org/maven2/com/firebaseui/firebase-ui-database/0.4.0/firebase-ui-database-0.4.0.pom
     Required by:
         project :app
   > Could not find com.miguelcatalan:materialsearchview:1.4.0.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/com/miguelcatalan/materialsearchview/1.4.0/materialsearchview-1.4.0.pom
       - https://repo.maven.apache.org/maven2/com/miguelcatalan/materialsearchview/1.4.0/materialsearchview-1.4.0.pom
     Required by:
         project :app

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
==============================================================================

5: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':app:mergeDebugJavaResource'.
> Could not resolve all files for configuration ':app:debugRuntimeClasspath'.
   > Could not find com.firebaseui:firebase-ui-database:0.4.0.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/com/firebaseui/firebase-ui-database/0.4.0/firebase-ui-database-0.4.0.pom
       - https://repo.maven.apache.org/maven2/com/firebaseui/firebase-ui-database/0.4.0/firebase-ui-database-0.4.0.pom
     Required by:
         project :app
   > Could not find com.miguelcatalan:materialsearchview:1.4.0.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/com/miguelcatalan/materialsearchview/1.4.0/materialsearchview-1.4.0.pom
       - https://repo.maven.apache.org/maven2/com/miguelcatalan/materialsearchview/1.4.0/materialsearchview-1.4.0.pom
     Required by:
         project :app

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
==============================================================================

6: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':app:checkDebugDuplicateClasses'.
> Could not resolve all files for configuration ':app:debugRuntimeClasspath'.
   > Could not find com.firebaseui:firebase-ui-database:0.4.0.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/com/firebaseui/firebase-ui-database/0.4.0/firebase-ui-database-0.4.0.pom
       - https://repo.maven.apache.org/maven2/com/firebaseui/firebase-ui-database/0.4.0/firebase-ui-database-0.4.0.pom
     Required by:
         project :app
   > Could not find com.miguelcatalan:materialsearchview:1.4.0.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/com/miguelcatalan/materialsearchview/1.4.0/materialsearchview-1.4.0.pom
       - https://repo.maven.apache.org/maven2/com/miguelcatalan/materialsearchview/1.4.0/materialsearchview-1.4.0.pom
     Required by:
         project :app

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
==============================================================================

7: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':app:desugarDebugFileDependencies'.
> Could not resolve all files for configuration ':app:debugRuntimeClasspath'.
   > Could not find com.firebaseui:firebase-ui-database:0.4.0.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/com/firebaseui/firebase-ui-database/0.4.0/firebase-ui-database-0.4.0.pom
       - https://repo.maven.apache.org/maven2/com/firebaseui/firebase-ui-database/0.4.0/firebase-ui-database-0.4.0.pom
     Required by:
         project :app
   > Could not find com.miguelcatalan:materialsearchview:1.4.0.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/com/miguelcatalan/materialsearchview/1.4.0/materialsearchview-1.4.0.pom
       - https://repo.maven.apache.org/maven2/com/miguelcatalan/materialsearchview/1.4.0/materialsearchview-1.4.0.pom
     Required by:
         project :app

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
==============================================================================

8: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':app:mergeDebugNativeLibs'.
> Could not resolve all files for configuration ':app:debugRuntimeClasspath'.
   > Could not find com.firebaseui:firebase-ui-database:0.4.0.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/com/firebaseui/firebase-ui-database/0.4.0/firebase-ui-database-0.4.0.pom
       - https://repo.maven.apache.org/maven2/com/firebaseui/firebase-ui-database/0.4.0/firebase-ui-database-0.4.0.pom
     Required by:
         project :app
   > Could not find com.miguelcatalan:materialsearchview:1.4.0.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/com/miguelcatalan/materialsearchview/1.4.0/materialsearchview-1.4.0.pom
       - https://repo.maven.apache.org/maven2/com/miguelcatalan/materialsearchview/1.4.0/materialsearchview-1.4.0.pom
     Required by:
         project :app

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
==============================================================================

* Get more help at https://help.gradle.org

BUILD FAILED in 3s
19 actionable tasks: 19 executed